lib/train/platforms/uuid.rb

```diff
@@ -0,0 +1,135 @@
+# encoding: utf-8
+
+require 'digest/sha1'
+require 'securerandom'
+require 'json'
+
+module Train::Platforms
+  class UUID
+    def initialize(platform)
+      @platform = platform
+      @backend = @platform.backend
+    end
+
+    def find_or_create_uuid
+      if @platform.unix?
+        unix
+      elsif @platform.windows?
+        windows
+      elsif @platform.name == 'aws'
+        aws
+      elsif @platform.name == 'azure'
+        azure
+      else
+        if @platform[:uuid_command]
+          result = @backend.run_command(@platform[:uuid_command])
+          return uuid_from_string(result.stdout.chomp) if result.exit_status == 0 && !result.stdout.empty?
+        end
+
+        raise 'Could not fine platform uuid! please set a uuid_command for your platform'
+      end
+    end
+
+    private
+
+    def aws
+      # parse the aws account id
+      client = @backend.aws_client(Aws::IAM::Client)
+      arn = client.get_user.user.arn
+      uuid_from_string(arn.split(':')[4])
+    end
+
+    def azure
+      # use subscription_id or tenant_id
```

Comment:
Redundant comment, yeah? Also the kind that becomes a lie when the implementation changes.